lib/src/revset.rs

```diff
@@ -2102,22 +2102,38 @@ impl SymbolResolver for FailingSymbolResolver {
 
 pub type PrefixResolver<'a, T> = Box<dyn Fn(&dyn Repo, &HexPrefix) -> PrefixResolution<T> + 'a>;
 
+pub trait PartialSymbolResolver<'a> {
```

Comment:
nit: generic lifetime `<'a>` isn't needed.

And `repo` could be passed to `resolve_symbol()` as an argument to eliminate lifetime at all, but I'm not sure if that make sense to the real extension impl.

Reply:
Ah, that's how that works.  Done.